meteor/iterative.py

```diff
@@ -176,7 +182,10 @@ def _iteratively_denoise_sf_amplitudes(
 
             # project onto the native amplitudes to obtain an "updated_derivative"
             #   Fh' = (D_F' + F) * [|Fh| / |D_F' + F|]
+            denoised_difference_sfs, native = filter_common_indices(denoised_difference_sfs, native)
```

Comment:
can you either
 - filter outside the while loop? or
 - calculate the indices used for filtration outside the loop?

I'm sure it doesn't affect the code's correctness, but it unnecessarily costly as written. Not a big deal if it's a hassle.

Reply:
Thanks for the explanation! I'm still a tad fuzzy on why you can't cache the indices of the offensive entries. Let's not get fixated on that though. Compared to the other ops in this loop, I would imagine the filtration is a couple of orders of magnitude faster. I'd say go ahead!